Database/CleanArchive.py

```diff
@@ -88,12 +96,69 @@ def clean(all=True):
         transaction = {
             '$pull': {'snippets': {'transcript': ''}}
         }
+        if duplicate < 1.0:
+            print('Duplicate found')
+            set_fields['duplicate'] = True
+            duplicates['metadata.Subtitle'] += 1
+        else:
+            set_fields['duplicate'] = False
         if errors:
             set_fields['errors'] = errors
         else:
             transaction['$unset'] = {'errors': 1}
         if set_fields:  # $set cannot be empty.
             transaction['$set'] = set_fields
+
+
+
         db.ArchiveIndex.update_one({'_id': episode['_id']}, transaction)
     print('Updates:', updates)
     print('Failures:', failures)
+    print('Duplicates', duplicates)
+
+
+def checkDuplicate(epi):
+    upperBound = db.ArchiveIndex.find({'identifier': epi['identifier']})[0]['date']
+    dateTimeObj = datetime.datetime.strptime(upperBound, '%Y-%m-%dT%H:%M:%SZ')
+    lowerBound = str(dateTimeObj - dateTimeObj - datetime.timedelta(days=1))
+    fields = db.ArchiveIndex.find({'date': {'$lte': upperBound, '$gte': lowerBound}}, {'snippets': 1})
+    currentEpisode = ''
+    for snip in epi['snippets']:
+        currentEpisode += str(snip['transcript'])
```

Comment:
Use the string join method.
' '.join([x['transcript'] for x in epi['snippets']])

Reply:
currentEpisode= ' '.join([x['transcript'] for x in epi['snippets']])
This results in every one being a duplicate for some reason

Reply:
Probably because somewhere it's comparing to an empty string or something.